solutions/cses-1641.mdx

```diff
@@ -51,3 +56,77 @@ int main(){
 	cout << "IMPOSSIBLE" << endl;
 }
 ```
+	
+</CPPSection>
+	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class SumOfThreeValues {
+	static class comparator implements Comparator<Group> {
+		public int compare (Group one, Group two) {
+			return Integer.compare(one.value, two.value);
+		}
+	}
+
+	static class Group {
```

Comment:
also what is this a group *of*
i have no idea what's happening rn

Reply:
its just a pair of input, should be clear after reading the problem. pair can be used but I don't really use that and a class is more clear

Reply:
where does the problem even mention pairs?

Reply:
> This should be the right implementation for extending `Comparable`. Make sure to try running it and seeing if it still compiles because I did this off the top of my head. I also added a constructor to your `Group` class because that's normally how you initialize objects in java.

I think I will revert to what I originally had I don't know what you're doing and how to make it compile

Reply:
it doesn't, it is a pair because you want to keep the input together, in a "pair'